neural_modelling/src/neuron/population_table/population_table_binary_search_impl.c

```diff
@@ -54,16 +57,32 @@ static inline uint32_t get_address(address_and_row_length entry) {
 }
 
 static inline uint32_t get_row_length(address_and_row_length entry) {
-    return entry & 0xFF;
+    // Row lengths are stored offset by 1, to allow 1-256 length rows
+    return (entry & 0xFF) + 1;
 }
 
 static inline uint32_t is_single(address_and_row_length entry) {
     return entry & 0x80000000;
 }
 
+static inline uint32_t get_n_neurons(master_population_table_entry entry) {
+    return entry.n_neurons_and_mask_shift >> 5;
+}
+
+static inline uint32_t get_core_shift(master_population_table_entry entry) {
+    return entry.n_neurons_and_mask_shift & 0x1F;
```

Comment:
another magic number. yet again, what happens with a retina. these feel hardcoded to just neuron models

Reply:
As above.